scripts/js/queries.js

```diff
@@ -40,39 +44,84 @@ function getDnssecConfig() {
   });
 }
 
+// Fetch database info (earliest timestamp, sizes, ...) from the API and
+// initialize related globals.
+function getDatabaseInfo() {
+  $.getJSON(document.body.dataset.apiurl + "/info/database", data => {
+    // earliest_timestamp is provided in seconds since epoch
+    // We have two sources: earliest_timestamp_disk (on-disk) and earliest_timestamp (in-memory)
+    // Use whichever is smallest and non-zero
+    const diskTimestamp = Number(data.earliest_timestamp_disk);
+    const memoryTimestamp = Number(data.earliest_timestamp);
+
+    // Filter out zero/invalid timestamps
+    const validTimestamps = [diskTimestamp, memoryTimestamp].filter(ts => ts > 0);
+
+    // Use the smallest valid timestamp, or null if none exist
+    beginningOfTime = validTimestamps.length > 0 ? Math.min(...validTimestamps) : null;
+
+    // Round down to nearest 5-minute segment (300 seconds) if valid
+    if (beginningOfTime !== null) {
+      beginningOfTime = Math.floor(beginningOfTime / 300) * 300;
+    }
+
+    // If from/until were not provided via GET, default them
+    // Only use defaults if beginningOfTime is valid
+    if (beginningOfTime !== null) {
+      from ??= beginningOfTime;
+      until ??= endOfTime;
+    }
+
+    initDateRangePicker();
+  });
+}
+
 function initDateRangePicker() {
+  // If there's no valid data in the database, disable the datepicker
+  if (beginningOfTime === null) {
+    $("#querytime").prop("disabled", true);
+    $("#querytime").addClass("disabled");
+    $("#querytime-note").text("No data in the database");
+    return;
+  }
+
+  const now = luxon.DateTime.now();
+  const minDateDt = luxon.DateTime.fromSeconds(beginningOfTime);
+  const maxDateDt = luxon.DateTime.fromSeconds(endOfTime);
+  const earliestDateStr = minDateDt.toFormat(dateformat);
+  $("#querytime-note").text(`Earliest date: ${earliestDateStr}`);
+
   $("#querytime").daterangepicker(
     {
       timePicker: true,
       timePickerIncrement: 5,
       timePicker24Hour: true,
-      locale: { format: dateformat },
-      startDate: luxon.DateTime.fromMillis(from * 1000), // convert to milliseconds since epoch
-      endDate: luxon.DateTime.fromMillis(until * 1000), // convert to milliseconds since epoch
+      locale: {
+        format: dateformat,
+        firstDay: 7,
```

Comment:
This might be a British thing, but my week starts on Monday
```suggestion
        firstDay: 1,
```

Reply:
I think this is more a personal choice.

Here, almost every calendar shows Sunday as the first column and starting a calendar with Monday is strange.

Reply:
We could implement something based on the local used by user's system
https://moment.github.io/luxon/#/intl?id=locale-based-weeks